IQSS/dataverse
# allow multiple required subfields #8236

**What this PR does / why we need it**:

Without a fix like the one in this pull request, as of Dataverse 5.4 (which contains pull request #7608), you cannot mark more than one subfield as required such as "Other ID Agency" and "Other ID Identifier" in the screenshot below.

![Screen Shot 2021-11-18 at 3 50 40 PM](https://user-images.githubusercontent.com/21006/142494622-0af509e2-00e0-4ab4-b0c0-32079dac8374.png)

**Which issue(s) this PR closes**:

Closes #8236

**Special notes for your reviewer**:

If using a set to ensure uniqueness is preferred, we can merge 291dbbe0d2 into this pull request. To compare that solution against develop: https://github.com/IQSS/dataverse/compare/8236-required-subfields-list2set

**Suggestions on how to test this**:

See the issue. My preferred way to test is the following:

- Create dataverse, save.
- Edit dataverse, change "Other ID Agency" to required (subfield of "Other ID"), save.
- Edit dataverse, change "Other ID Identifier" as required (subfield of "Other ID"), save.

On develop or https://demo.dataverse.org you get this:

- BOOM! PSQLException: ERROR: duplicate key value violates unique constraint "unq_dataversefieldtypeinputlevel_0"

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

No.

**Is there a release notes update needed for this change?**:

I don't think so. It's just a bug fix.

**Additional documentation**:

None.
